File: bench/code/fib/fib.c
#include <stdio.h>

int fib(int n) {
  if (n <= 1) {
    return n;
   } else {
      return fib(n-1) + fib(n-2);
   }
}

int main() {
   int n = 35;
   int r = fib(n);
   printf("%d: %d\n", n, r);
}

File: bench/code/fib/fib.go
package main

import (
	"C" // without this line, Go toolchain does not work. It is a bug in the Golang compiler
	"fmt"
)

func fib(n uint) uint {
	if n == 0 {
		return 0
	} else if n == 1 {
		return 1
	} else {
		return fib(n-1) + fib(n-2)
	}
}

func main() {
	n := uint(40)
	fmt.Println(fib(n))
}
